ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -331,17 +345,16 @@ def close(self) -> None:
         """
         pass
 
+    @property
     @abstractmethod
-    def get_behavior_names(self) -> List[BehaviorName]:
+    def behavior_specs(self) -> BehaviorMapping:
         """
-        Returns the list of the behavior names present in the environment.
+        Returns a Mapping from behavior names to behavior specs.
         Agents grouped under the same behavior name have the same action and
         observation specs, and are expected to behave similarly in the
         environment.
-        This list can grow with time as new policies are instantiated.
-        :return: the list of agent BehaviorName.
+        Note that this mapping can grow and change as new policies are instantiated.
```

Comment:
"and change" - the spec that a behavior name is associated with should always be the same, right?

Maybe "New keys can be added to this mapping as new policies are instantiated"?